### ISSUE TITLE
[BUG] macos service.running is not robust

### ISSUE THREAD

**Issue by noelmcloughlin** [saltstack/salt#57664]
**Description**
MacOS `service.running` is not robust.  Testing https://github.com/saltstack-formulas/salt-formula/pull/447  I only managed to get `com.saltstack.salt.master` success once, every other attempt returns-

1. `/Library/LaunchDaemons/com.saltstack.salt.master.plist: service already loaded`
2. `Service com.saltstack.salt.master is already enabled, and is dead`
3. `/Library/LaunchDaemons/com.saltstack.salt.master.plist: Service is disabled`

**Setup**
(Please provide relevant configs and/or SLS files (be sure to remove sensitive info).

**Steps to Reproduce the behavior**
LOGS
```
$ sudo launchctl load /Library/LaunchDaemons/com.saltstack.salt.master.plist && sudo launchctl unload /Library/LaunchDaemons/com.saltstack.salt.master.plist

$ sudo salt-call state.highstate --local

          ID: salt-master
    Function: service.running
        Name: com.saltstack.salt.master
      Result: False
     Comment: Service com.saltstack.salt.master is already enabled, and is dead
     Started: 21:35:17.608376
    Duration: 302.159 ms
     Changes:   
              ----------
              com.saltstack.salt.master:



$ sudo launchctl load /Library/LaunchDaemons/com.saltstack.salt.master.plist
$ sudo salt-call state.highstate --local

          ID: salt-master
    Function: service.running
        Name: com.saltstack.salt.master
      Result: False
     Comment: Failed to bootstrap service:
              stdout: 
              stderr: /Library/LaunchDaemons/com.saltstack.salt.master.plist: service already loaded
              retcode: 37
     Started: 21:41:34.882255
    Duration: 283.856 ms
     Changes:   


$ sudo launchctl unload /Library/LaunchDaemons/com.saltstack.salt.master.plist
$ sudo rm /Library/LaunchDaemons/com.saltstack.salt.master.plist
$ sudo salt-call state.highstate --local


[INFO    ] Running state [com.saltstack.salt.master] at time 21:59:57.950310
[INFO    ] Executing state service.running for [com.saltstack.salt.master]
[DEBUG   ] LazyLoaded mac_utils.available_services
[DEBUG   ] Loading available services
[DEBUG   ] read_plist: Gathering service info for /Library/LaunchDaemons/com.saltstack.salt.master.plist
[DEBUG   ] Loading available services
[DEBUG   ] Found context for available services.
[DEBUG   ] Loading available services
[DEBUG   ] Found context for available services.
[DEBUG   ] Loading available services
[DEBUG   ] Found context for available services.
[DEBUG   ] Loading available services
[DEBUG   ] Found context for available services.
[DEBUG   ] Loading available services
[DEBUG   ] Found context for available services.
[DEBUG   ] Loading available services
[DEBUG   ] Found context for available services.
[ERROR   ] Failed to bootstrap service:
stdout:
stderr: /Library/LaunchDaemons/com.saltstack.salt.master.plist: Service is disabled
retcode: 119
[INFO    ] Completed state [com.saltstack.salt.master] at time 21:59:58.353670 (duration_in_ms=403.359)


          ID: salt-master
    Function: service.running
        Name: com.saltstack.salt.master
      Result: False
     Comment: Failed to bootstrap service:
              stdout: 
              stderr: /Library/LaunchDaemons/com.saltstack.salt.master.plist: Service is disabled
              retcode: 119
     Started: 22:15:40.984792
    Duration: 282.772 ms
     Changes: 

```

**Expected behavior**

**Screenshots**

**Versions Report**
<details><summary>salt --versions-report</summary>
(Provided by running salt --versions-report. Please also mention any differences in master/minion versions.) 

```
Salt Version:
           Salt: 3000.3
 
Dependency Versions:
           cffi: 1.12.2
       cherrypy: 17.4.1
       dateutil: 2.8.0
      docker-py: Not Installed
          gitdb: 2.0.6
      gitpython: 2.1.15
         Jinja2: 2.10.1
        libgit2: 1.0.0
       M2Crypto: 0.35.2
           Mako: 1.0.7
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: 2.19
       pycrypto: 3.8.1
   pycryptodome: Not Installed
         pygit2: 1.2.1
         Python: 3.7.7 (default, Mar 10 2020, 15:43:03)
   python-gnupg: 0.4.4
         PyYAML: 5.1.2
          PyZMQ: 18.0.1
          smmap: 3.0.4
        timelib: 0.2.4
        Tornado: 4.5.3
            ZMQ: 4.3.1
 
System Versions:
           dist:   
         locale: UTF-8
        machine: x86_64
        release: 19.5.0
         system: Darwin
        version: 10.15.5 x86_64
```
</details>

**Additional context**

**Comment by weswhet** [comment]
Ah looks like you are running into 2 different issues. That I've been meaning to fix. 

1. The service state will attempt to start macOS service before enabling it. on macOS, you can't `load/bootstrap/start` the service if it's disabled. So the state will always fail to start a disabled service macOS. The fix here is to enable services before starting them on macOS.

2. The `mac_service.enable` function is really doing a `launchctl list` which isn't a good way to check if a service is enabled/disabled. We should do a `not disabled()` call as the `mac_service.disabled` function is actually doing the right thing here.

**Comment by weswhet** [comment]
The issue when you get a `stderr: /Library/LaunchDaemons/com.saltstack.salt.master.plist: service already loaded` isn't an issue per se with the salt service module. It's with the salt master daemon. When checking to see if a service is running on macOS it will check the launchd config for that service to see if it's supposed to be an always running service. AKA a service that's supposed to have an active PID. If it's supposed to be active and it's NOT, salt will try to reload it. When it does attempt to reload/start you get the above already loaded error. We could probably better handle this by completely stopping the service first and then starting it again. Expect this doesn't really handle the underlying issue of your service not working properly it just keeps salt from Erroring/Failing. This behavior is a good indicator that there is an issue with your service. I would enable the `launchd` service logging (found in `com.saltstack.salt.master.plist`) and go from there.

**Comment by noelmcloughlin** [comment]
- I enabled debug but nothing was logged.  
- I did `launchctl dumpstate` and noticed the service was disabled=True.

The file `https://raw.githubusercontent.com/saltstack/salt/master/pkg/osx/scripts/com.saltstack.salt.master.plist` mentioned in https://github.com/saltstack-formulas/salt-formula/pull/447 references `/opt/salt/bin/start-salt-master.sh` but that does not exist on my filesystem, so 

- I created a dummy `/opt/salt/bin/start-salt-master.sh` file to satisfy the service.
- next, I did the following
```
sudo launchctl enable system/com.saltstack.salt.master
sudo launchctl unload /Library/LaunchDaemons/com.saltstack.salt.master.plist
sudo rm /Library/LaunchDaemons/com.saltstack.salt.master.plist
```

Finally the service.running state worked for `com.saltstack.salt.master`

**Comment by xeacott** [comment]
Thanks for the detailed report @noelmcloughlin / @weswhet Is it okay if I leave this issue tagged as it is and we continue to offer support? Or is there another ticket we can link this to?

**Comment by sagetherage** [comment]
This was fixed with https://github.com/saltstack/salt/blob/master/salt/modules/mac_brew_pkg.py#L30-L38
